### PR TITLE
add utterances

### PR DESCRIPTION
**Pull request by coder-lg**
So, people who know this know that Disqus is full of ads if you use it for free, and Cusdis might nag people like me for permission. So, I found out about Utterances, another chat system and it is open source, and uses GitHub login. I am trying to add it here. There has to be a mistake. If you find it, please change it.

Changes made to files:
config.yml: added utterances permission code and a whole new set of code to choose the comment system.

New files:
utterances.html ~~~> here has to be a mistake, please change it

### CODE REVIEW

**Comment by coder-lg** [comment]
@sylhare and, forgot to mention, it saves the comments as GitHub issues in the repo

**Comment by sylhare** [comment]
How I did not know about this one, I'll test it out ;)

**Comment by sylhare** [comment]
Ok I see what you did there, it's a bit raw but I think it's a good idea to have a `comments provider` field and then choose between the one you want 🤔

**Comment by coder-lg** [comment]
Is it good?

**Comment by coder-lg** [comment]
> Ok I see what you did there, it's a bit raw but I think it's a good idea to have a `comments provider` field and then choose between the one you want 🤔 
> 

The comments provider was in another user, LazyRen's blog, I found it in Google when I searched for a regulation code for  the comment systen in it.

**Comment by coder-lg** [comment]
Does it work like you expected?

**Comment by danisztls** [comment]
@coder-lg You have to modify the post layout to parse the environment variables you wrote at config and include the include you wrote.

**Comment by coder-lg** [comment]
Can you do it, because I don't know how to change that.

**Comment by danisztls** [comment]
> Can you do it, because I don't know how to change that.

Don't have the time right now.

**Comment by coder-lg** [comment]
No worries, you can do it anytime

**Comment by coder-lg** [comment]
@danisztls I know how I can edit the post layout with minimal damage. I need to change cusdis statement to elseif and utterances will be else. I will make a new PR within 48 hours. Not closing this yet, will close after I make the new PR

**Comment by coder-lg** [comment]
I got some kind of idea. Why not we add the Utterances code into the post.html file directly with an if statement?

**Comment by coder-lg** [comment]
And also, there is a typo in post.html where Utterances is accidentally named Utternaces, I will change it today.

**Comment by coder-lg** [comment]
I think you have to change all the things yourself, sorry, but I can't add Utterances here. I will add it to my personal blog though. It uses this same theme.

**Comment by coder-lg** [comment]
I think you have to change all the things yourself, sorry, but I can't add Utterances here. I will add it to my personal blog though. It uses this same theme.

File: _layouts/post.html [hunk 1]
```diff
@@ -44,6 +44,9 @@ <h1 id="{{ page.title | cgi_escape }}" class="title">{{ page.title }}</h1>
 {% if site.disqus_shortname or site.theme_settings.disqus_shortname %}
 {% include social/disqus.html %}{% endif %}
 
+<!--utterances-->
+{% if site.utterances %}{% include social/utternaces.html %}{% endif %}
```
Comment (danisztls) [inline _layouts/post.html:48]: typo

File: _includes/social/utterances.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<script src="https://utteranc.es/client.js"
+        repo="sylhare/Type-on-Strap"
+        issue-term="pathname"
+        theme="dark"
```
Comment (danisztls) [inline _includes/social/utterances.html:4]: You should use Liquid variables as you created them at `_config`.
Eg. `theme="dark"` should be `theme="{{ site.utterances.theme }}"`

File: _config.yml [hunk 3]
```diff
@@ -25,7 +25,27 @@ color_image: /assets/img/lineart.png                    # A bit transparent for
...
+  
+# You must install utterances github app before use.(https://github.com/apps/utterances)
+# Make sure all variables are set properly. Check below link for detail.
+# https://utteranc.es/
+# for your information, 
+# utterances is open source and free.
+# no ads at all where disqus provides ads and no permission needed when you are using cusdis
+utterances:
+  repo:                "[your repo name here]"
+  issue-term:          "pathname"
```
Comment (danisztls) [inline _config.yml:44]: What `issue-term` and `label` do?

File: _layouts/post.html (hunk 1 above)
Comment (coder-lg) [reply _layouts/post.html:48]: Yep, Daniel

File: _config.yml (hunk 3 above)
Comment (coder-lg) [reply _config.yml:44]: Issue term is, first, Utterances store the comments as GitHub issues. So, the issue term can be configured in Utterances' official [site](https://utteranc.es). The term can be the title of the blog, or more. Labels, are like the issue labels too.

**Comment by coder-lg** [comment]
Actually, put the Utterances code inside post.html because, it has to be in a specific layout. The place is between `</div>` and `</script>`.

**Pull request closed without merging**